### DISCUSSION TITLE
Turbo Pascal Version?

### DISCUSSION THREAD

**Discussion by seljuck** [magiblot/tvision#66] [General]
Just curious if there's a way to make this work with the Pascal Version of Turbo Vision?

I have an old Turbo Pascal Program I'd like to update to run on Windows 10.

**Comment by magiblot** [comment]
Hi Daniel!

No, I'm sorry. This project can't be used with Pascal applications out-of-the-box. It is necessary to either rewrite that application in C++ or rewrite this project in Pascal.

**Reply by seljuck** [reply]
Thanks for the quick response.

**Comment by mooskagh** [comment]
There was an attempt to implement Pascal version of Turbo Vision in [this project](https://wiki.freepascal.org/Free_Vision), but I guess it's not maintained. (although the German tutorial mentioned on that page is pretty fresh, so maybe it has some life).

The problem with the original Pascal Turbo Vision is that unlike C++ version, it was never open sourced, and Free Vision was likely an attempt to implement Pascal TVision specification by looking in C++ TVision source code.

As a matter of fact, C++ itself was allegedly written as a port of Pascal Turbo Vision, kind of opposite of what Free Vision attempted to do.

**Reply by seljuck** [reply]
Thanks for the info. I took a quick look. It sound promising. Maybe I'll be lucky and I can just recompile my code with no changes. lol.
